lib/matplotlib/offsetbox.py

```diff
@@ -91,15 +91,14 @@ def _get_packed_offsets(wd_list, total, sep, mode="fixed"):
     offsets : array of float
         The left offsets of the boxes.
     """
-    w_list, d_list = zip(*wd_list)
-    # d_list is currently not used.
+    w_list, d_list = zip(*wd_list)  # d_list is currently not used.
+    cbook._check_in_list(["fixed", "expand", "equal"], mode=mode)
 
     if mode == "fixed":
         offsets_ = np.cumsum([0] + [w + sep for w in w_list])
         offsets = offsets_[:-1]
         if total is None:
             total = offsets_[-1] - sep
-        return total, offsets
```

Comment:
I have a slight preference for keeping these. It's simpler to follow the code when returning early.

Reply:
I wouldn't have made the change except for the fact that the function immediately below (_get_aligned_offsets) uses single-return-at-end-of-if-elif-block, so I made both of them consistent.

Reply:
The difference is that the function below has typically only two lines per case which are just assignments (not much going on there), but here we have 4-9 lines with conditional logic per case. IMHO that justifies different approaches.

Reply:
ok, restored